### PR TITLE
149 column rename

### LINKED ISSUE #149: [ALIASES] Fix column renaming for normal users
# STR:

1. Normal user (no `aliases` feature flag) uploads dataset
2. Create map with dataset
3. Get into table view in created map
4. Rename column

# Expected behaviours:

Column gets renamed

# Actual behaviour

No popup for rename appears.

### PR DESCRIPTION
**Pull request by coyle5280**
This closes #149 

# Changes
Updated `table.js` so that when a data-user renames a column the corresponding alias column name object is updated as well
Updated  `table_header_view.jst.ejs` to handle new logic for column renaming
Updated `tabledata.js` to not place column name in alias field. Alias field is now a null value.
Updated `wizard.js` to handle a null column alias value
Updated `header_view.js` to use alias value as column name if available if not to use original column name. Also added field for original column name so that a datauser can see old and new name.   

# Acceptance
- [ ] Regular user should be able to edit column name if table was owned by user and not a remote or imported table
- [ ] Regular user should only see column name and type.  Column name should reflect alias value if table has been imported and the column has been given an alias. If not user should see original column name.
- [ ] Data-user can see both original and alias column name
- [ ] Data-user can edit both alias column name and original column name
- [ ] Column alias values can only be created or edited in 'Edit Mode'.  
- [ ] Data-user can still see original and alias column names in 'regular mode' but cannot edit alias name only original name
- [ ] If data-user edits a column name and that column has an alias, the new alias key-value is add and the old key-value is removed.

### CODE REVIEW

File: vendor/assets/javascripts/select2.min.js [hunk 1]
```diff
@@ -833,8 +833,8 @@ the specific language governing permissions and limitations under the Apache Lic
                     process=function(element, collection) {
                         var group;
                         if (element.is("option")) {
-                            if (query.matcher(term, (element.attr('alias')) ? element.attr('alias') : element.text(), element)) {
-                                collection.push({id:element.attr("value"), text:element.text(), element: element.get(), css: element.attr("class"), alias: element.attr('alias')});
+                            if (query.matcher(term, (element.attr('alias') !== 'undefined') ? element.attr('alias') : element.text(), element)) {
```
Comment (gfiorav) [inline vendor/assets/javascripts/select2.min.js]: I'm not sure we can modify JQuery library (licensing).

We'll have to use another way :)

**Review by gfiorav — CHANGES_REQUESTED** [review]
Let's discuss this approach

File: lib/assets/javascripts/cartodb/models/wizard.js [hunk 2]
```diff
@@ -88,7 +88,7 @@ cdb.admin.FormSchema = cdb.core.Model.extend({
 
             // Used to sort be Alias first.
             columns = _.sortBy(columns, function (r) {
-                return r[2].toLowerCase();
+                return (r[2]) ? r[2].toLowerCase() : r[0].toLowerCase();
```
Comment (gfiorav) [inline lib/assets/javascripts/cartodb/models/wizard.js]:
```js
return (r[2] || r[0]).toLowerCase();
```

Comment (coyle5280) [reply lib/assets/javascripts/cartodb/models/wizard.js]: This has been updated

**Review by gfiorav — APPROVED** [review]
Thanks!

**Comment by coyle5280** [comment]
The changes have been made to correct the alias add vs edit bug that was occurring.  Also I updated the empty alias value to be null and not undefined.  When adding an alias the datauser should now see the add option and not the edit/remove option unless an alias was already added.

File: lib/assets/javascripts/cartodb/table/header_dropdown.js [hunk 3]
```diff
@@ -50,11 +50,7 @@ cdb.admin.HeaderDropdown = cdb.admin.DropdownMenu.extend({
   setTable: function (table, column, alias) {
     this.table = table;
     this.column = column;
-    if (column !== alias) {
-      this.options.columnAlias = alias;
-    } else {
-      this.options.columnAlias = null;
-    }
+    this.options.columnAlias = alias;
```
Comment (coyle5280) [inline lib/assets/javascripts/cartodb/table/header_dropdown.js:53]: Removed redundant code.

**Review by gfiorav — APPROVED** [review]
Thanks!

**Pull request merged**